qiskit/quantum_info/synthesis/two_qubit_decompose.py

```diff
@@ -926,9 +948,237 @@ def __call__(self, target, basis_fidelity=None, *, _num_basis_uses=None) -> Quan
             return_circuit.append(self.gate, [q[0], q[1]])
         return_circuit.compose(decomposition_euler[2 * best_nbasis], [q[0]], inplace=True)
         return_circuit.compose(decomposition_euler[2 * best_nbasis + 1], [q[1]], inplace=True)
-
         return return_circuit
 
+    def _pulse_optimal_chooser(self, best_nbasis, decomposition, target_decomposed):
+        """Determine method to find pulse optimal circuit. This method may be
+        removed once a more general approach is used.
+
+        Returns:
+            QuantumCircuit: pulse optimal quantum circuit.
+            None: Probably nbasis=1 and original circuit is fine.
+
+        Raises:
+            QiskitError: Decomposition for selected basis not implemented.
+        """
+        circuit = None
+        if self.pulse_optimize and best_nbasis not in {2, 3}:
+            raise QiskitError(
+                f"Unexpected number of entangling gates ({best_nbasis}) in decomposition."
+            )
```

Comment:
I'd prefer not to raise in this case, rather it seems preferable to consider nbasis=0 or 1 as already-optimal.

Reply:
Otherwise if you set `pulse_optimize=True` at the pass level, then you get an error if any SU(4) has a 1-CNOT expansion, whereas presumably your intent was to get 0,1,2,3 CNOT implementations where the 2,3 CNOT expansions are pulse optimized.

Reply:
thanks for catching that